compiler/src/iree/compiler/InputConversion/StableHLO/Preprocessing/test/stablehlo_to_stablehlo.mlir

```diff
@@ -462,3 +462,12 @@ func.func @broadcast_iota_sort_slice_incorrect_dims(%in : tensor<16x16x16xf32>)
 
 // CHECK-LABEL: @broadcast_iota_sort_slice_incorrect_dims
 // CHECK-NOT: chlo.top_k
+
+// CHECK-LABEL: @concat_remove_zero_extents
+func.func @concat_remove_zero_extents(%arg0: tensor<2x3xi32>, %arg1 : tensor<2x3xi32>, %arg2 : tensor<2x0xi32>) -> tensor<2x6xi32> {
+  %0 = stablehlo.concatenate %arg0, %arg1, %arg2, dim = 1 : (tensor<2x3xi32>, tensor<2x3xi32>, tensor<2x0xi32>) -> tensor<2x6xi32>
+  // CHECK-NOT: stablehlo.concatenate %arg0, %arg1, %arg2
+  // CHECK: [[R0:%.+]] = stablehlo.concatenate %arg0, %arg1, dim = 1 : (tensor<2x3xi32>, tensor<2x3xi32>) -> tensor<2x6xi32>
```

Comment:
We can capture `arg0` and `arg1` like below example, and check that the result of `concat` is returned. Then we don't need `CHECK-NOT` anymore

https://github.com/openxla/iree/blob/554d40a3130160730a62e8e936904cc2a1e4405b/compiler/src/iree/compiler/Dialect/Util/IR/test/hint_ops.mlir#L3-L7

Reply:
Updated